### ISSUE TITLE
failed Python3 initialization

### ISSUE THREAD

**Issue by dberansky** [twosigma/beakerx#4034]
I've been using Beaker just fine for a few days, but earlier in the day, I started getting an error message that Python3 cannot be initialized.  The only error I see in the log is this (repeated multiple times):

> nginx-stderr>2016/04/26 23:16:35 [error] 23225#0: *57 connect() failed (111: Connection refused) while connecting to upstream, client: 10.64.103.195, server: vm-01, request: "GET /34d512f/python3.238524596387/login HTTP/1.1", upstream: "http://127.0.0.1:8804/login", host: "vm-01:8800", referrer: "https://vm-01:8800/beaker/"

As I understand, the upstream connection on port 8804 is where iPython runs.  But if I try to manually connect to the port, the connection works:

> (/data/dm-conda)dmitryb@vm-01:~$ telnet 127.0.0.1 8804
> Trying 127.0.0.1...
> Connected to 127.0.0.1.
> Escape character is '^]'.

How else can I troubleshoot this issue?

**Comment by scottdraves** [comment]
The big question is what changed between when it worked and when it does not?
you are on linux?  did you install anything?

**Comment by scottdraves** [comment]
also, open the browser's debug console and see if any errors are reported there.

**Comment by dberansky** [comment]
Yes, that is a big question and that was the first thing I considered, but as far as I am aware there were no changes on the backend.  I'm running Beaker on a dedicated VM instance of Ubuntu  (exactly for the reason of keeping the environment stable).  I started getting the message in the middle of working on the notebook's content (actually, after I switched from being docked at work to VPN over wifi at home).

As of this morning, these are the errors in the browser console when I try to reload the notebook:

> Failed to load resource: the server responded with a status of 403 (Forbidden)
> python3.js:141 Uncaught TypeError: Cannot read property 'execute' of undefined
> python3.js:153 TIMED OUT - waiting for ipython kernel to startspin @ python3.js:153
> python3.js:141 Uncaught TypeError: Cannot read property 'execute' of undefined

and there are additional errors, it seems, in the server log:

> plugin service Python3 already started atpython3.803056714861
> Python3-stderr>[I 08:33:46.417 NotebookApp] 302 POST /login?next=%2F (127.0.0.1) 1.01ms
> Python3-stderr>[W 08:33:46.472 NotebookApp] 403 POST /api/sessions (127.0.0.1) 1.96ms referer=https://vm-01:8800/beaker/
> plugin service Python3 already started atpython3.803056714861
> Python3-stderr>[I 08:36:20.774 NotebookApp] 302 POST /login?next=%2F (127.0.0.1) 0.99ms
> Python3-stderr>[W 08:36:20.848 NotebookApp] 403 POST /api/sessions (127.0.0.1) 2.58ms referer=https://vm-01:8800/beaker/

**Comment by dberansky** [comment]
It appears that the browser errors are slightly different if I open the notebook after restarting the beaker server:

> **Failed to load resource: the server responded with a status of 502 (Bad Gateway)**
> https://vm-01:8800/b7c81a9/python3.413488880147/api/sessions/ Failed to load resource: the server responded with a status of 403 (Forbidden)
> python3.js:141 Uncaught TypeError: Cannot read property 'execute' of undefined
> python3.js:153 TIMED OUT - waiting for ipython kernel to startspin @ python3.js:153
> 3python3.js:141 Uncaught TypeError: Cannot read property 'execute' of undefined

**Comment by scottdraves** [comment]
what does jupyter --version say?

**Comment by dberansky** [comment]
```
$ which jupyter
/data/dm-conda/bin/jupyter
$ jupyter --version
4.1.0
```

**Comment by dberansky** [comment]
Found discussion #1952  with similar symptoms.  One recommendation was to try using Firefox.  So I did and surely enough things are working there just fine

**Comment by scottdraves** [comment]
try an incognito window in chrome

**Comment by dberansky** [comment]
Oh, crap.  Ok, clearing cookies was the first thing I tried, but I only cleared cookies for my server's domain.  I left company-wide cookies in place.  I should've known better.  Our IT is using a federated Single-Signon solution called SiteMinder and it looks like its authentication cookie has expired and somehow was blocking requests (I'm saying "somehow" because the Beaker server isn't SiteMinder protected, so I'm not sure why Beaker was affected, but clearing all cookies did the trick).
